src/TestSuite/TestCase.php

```diff
@@ -673,12 +673,12 @@ protected function skipUnless($condition, $message = '')
      * Mock a model, maintain fixtures and table association
      *
      * @param string $alias The model to get a mock for.
-     * @param array $methods The list of methods to mock
+     * @param array|null $methods The list of methods to mock
      * @param array $options The config data for the mock's constructor.
      * @throws \Cake\ORM\Exception\MissingTableClassException
      * @return \Cake\ORM\Table|\PHPUnit_Framework_MockObject_MockObject
      */
-    public function getMockForModel($alias, array $methods = [], array $options = [])
+    public function getMockForModel($alias, $methods = [], array $options = [])
```

Comment:
removing a typehint sounds wrong - why not `array $methods = null`?

Reply:
We usually do not allow an additional type `null` where not necessary, `[]` is the same and better in these cases.
So I would suggest we do not change this here for no reason.

Reply:
Sorry, but they are completely different. With an empty array all methods are stubs and all return `null` by default; with `null`, all methods are mocks and they run the original code contained within the method when called.